src/figuresClasses.ts

```diff
@@ -1,19 +1,80 @@
-export interface Figure {
 
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Negative value entered');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('Impossible triangle');
```

Comment:
It would be great to specify why exactly it's not possible
![image](https://user-images.githubusercontent.com/96409563/228446787-3f5214c8-72dc-4f44-a51a-fd3d4278ea29.png)